src/printer/smt2/smt2_printer.cpp

```diff
@@ -252,11 +252,36 @@ void Smt2Printer::toStream(std::ostream& out,
     }
 
     case kind::DATATYPE_TYPE:
+    {
+      const Datatype& dt = (NodeManager::currentNM()->getDatatypeForIndex(
+          n.getConst<DatatypeIndexConstant>().getIndex()));
+      if (dt.isTuple())
+      {
+        unsigned int n = dt[0].getNumArgs();
+        if (n == 0)
+        {
+          out << "Tuple";
+          for (unsigned int i = 0; i < n; i++)
```

Comment:
You can remove these 4 lines since n=0.

Reply:
You're right, the loop is unnecessary for the empty tuple case. Thanks.